v1/fs/index.ts

```diff
@@ -2,37 +2,45 @@ import path from 'path'
 import fs from 'fs'
 import tar from 'tar-fs'
 import gunzip from 'gunzip-maybe'
+import rimraf from 'rimraf'
+import { BusboyFileStream } from "@fastify/busboy";
 
 export class SiteFileSystem {
   path: string
-  constructor (rootPath: string) {
+  constructor(rootPath: string) {
     this.path = path.join(rootPath, 'sites')
   }
 
-  async clear (siteId: string): Promise<void> {
+  async clear(siteId: string): Promise<void> {
     const sitePath = this.getPath(siteId)
-    fs.rmSync(sitePath, { recursive: true, force: true })
+    return await new Promise((resolve, reject) => rimraf(sitePath, (err) => {
+      if (err === null || err === undefined) {
+        resolve()
+      } else {
+        reject(err)
+      }
+    }))
   }
 
-  getPath (siteId: string): string {
+  getPath(siteId: string): string {
     return path.join(this.path, siteId)
   }
 
   /// Reads a .tar or .tar.gz from given `tarballPath` and extracts it to
   /// the target directory. Deletes original tarball when done
-  async extract (tarballPath: string, siteId: string): Promise<void> {
+  async extract(stream: BusboyFileStream, siteId: string): Promise<void> {
     return await new Promise((resolve, reject) => {
       const sitePath = this.getPath(siteId)
-      fs.createReadStream(tarballPath).on('error', reject)
+      stream.on('error', reject)
```

Comment:
I'd suggest using the `pipeline` function instead of piping and adding callbacks.

```javascript
import { pipeline } from 'node:stream/promises';

await pipeline(
  createReadStream('archive.tar'),
  createGzip(),
  createWriteStream('archive.tar.gz'),
);
```

https://nodejs.org/api/stream.html#streampipelinesource-transforms-destination-options